code/datums/components/connect_loc_behalf.dm

```diff
@@ -0,0 +1,85 @@
+/// This element behaves the same as connect_loc, hooking into a signal on a tracked object's turf
+/// It has the ability to react to that signal on behalf of a seperate listener however
+/// This has great use, primarially for components, but it carries with it some overhead
+/// So we do it seperately rather then intigrating the behavior with the main element
+/datum/component/connect_loc_behalf
+	dupe_mode = COMPONENT_DUPE_SELECTIVE
+
+	datum_flags = DF_SIGNAL_VERBOSE
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+
+	var/atom/movable/tracked
+
+	var/atom/signal_atom
+
+/datum/component/connect_loc_behalf/Initialize(atom/movable/tracked, list/connections)
+	. = ..()
+	if (!istype(tracked))
+		return COMPONENT_INCOMPATIBLE
+	src.connections = connections
+	src.tracked = tracked
+
+/datum/component/connect_loc_behalf/CheckDupeComponent(datum/component/component, atom/movable/tracked, list/connections)
+	if(src.tracked != tracked)
+		return
+
+	// Not equivalent. Checks if they are not the same list via shallow copy.
+	if(src.connections ~! connections)
```

Comment:
I don't really like this operator, thought this was list operator abuse for a sec honestly